File: copy_files.js
const fs = require('fs');
const path = require('path');

function copyRecursiveSync(src, dest) {
    const exists = fs.existsSync(src);
    const stats = exists && fs.statSync(src);
    const isDirectory = exists && stats.isDirectory();
    if (isDirectory) {
        if (!fs.existsSync(dest)) {
            fs.mkdirSync(dest, { recursive: true });
        }
        fs.readdirSync(src).forEach((childItemName) => {
            copyRecursiveSync(path.join(src, childItemName), path.join(dest, childItemName));
        });
    } else {
        fs.copyFileSync(src, dest);
    }
}

const buildWeb = path.join(__dirname, 'build', 'web');
const publicTuktuk = path.join(__dirname, 'public', 'tuktuk');

console.log(`Copying from ${buildWeb} to ${publicTuktuk}...`);
try {
    copyRecursiveSync(buildWeb, publicTuktuk);
    console.log('Copy successful!');
} catch (err) {
    console.error('Copy failed:', err);
    process.exit(1);
}

File: copy_files.py
import shutil
import os

src = r"d:\Flutterapp\caculateapp\build\web"
dst = r"d:\Flutterapp\caculateapp\public\tuktuk"

print(f"Copying from {src} to {dst}...")

if not os.path.exists(src):
    print(f"Error: Source {src} does not exist!")
    exit(1)

# Remove destination if it exists
if os.path.exists(dst):
    print(f"Destination {dst} exists, removing...")
    try:
        # Use a more aggressive removal for Windows
        import subprocess
        subprocess.run(['rmdir', '/s', '/q', dst], shell=True)
    except Exception as e:
        print(f"Failed to remove dst via rmdir: {e}")
        # Fallback to shutil
        shutil.rmtree(dst, ignore_errors=True)

try:
    shutil.copytree(src, dst)
    print("Copy completed successfully!")
    
    # Verify
    files = os.listdir(dst)
    print(f"Files in {dst}: {files}")
    
except Exception as e:
    print(f"Error during copy: {e}")
    exit(1)
